### ISSUE TITLE
Better command line interface for running unit tests

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#1074]
The current command line UI for running unit tests is clunky. We could improve it by a lot. Some ideas:
- Have a separate runner for unit tests only, such as `unittests.py` or `tests.py`.
- Have it accept regexps instead of glob patterns, and perform a search instead of a full string match.

With these changes, running all test cases with `foobar` in their name would be like this:

```
$ ./tests.py foobar
```

Currently this is needed:

```
$ ./runtests.py unit-test -a "*foobar*"
```

**Comment by gvanrossum** [comment]
I also wish the unit tests weren't in a subdirectory of the mypy package.

In https://github.com/python/asyncio I have the following structure:

- There's a toplevel 'tests' folder containing unittests as files named
test_foobar.py

- The toplevel runtests.py script by default runs all tests it can find,
and takes optional pattern arguments that are matched (searched, actually)
against the test id (test IDs are the canonical test names, e.g.
'test_filename.TestCaseClass.test_method'). It also takes -x arguments that
are patterns to exclude. And a lot of others. The only thing it doesn't do
is run tests in parallel or run things other than unittests.

On Sun, Dec 13, 2015 at 12:31 PM, Jukka Lehtosalo <notifications@github.com>
wrote:

> The current command line UI for running unit tests is clunky. We could
> improve it by a lot. Some ideas:
>
>    - Have a separate runner for unit tests only, such as unittests.py or
>    tests.py.
>    - Have it accept regexps instead of glob patterns, and perform a
>    search instead of a full string match.
>
> With these changes, running all test cases with foobar in their name
> would be like this:
>
> $ ./tests.py foobar
>
> Currently this is needed:
>
> $ ./runtests.py unit-test -a "*foobar*"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JukkaL/mypy/issues/1074>.
>



-- 
--Guido van Rossum (python.org/~guido)

**Comment by JukkaL** [comment]
That all sounds pretty reasonable. Another idea is to rename the current test script to something like `ci_tests.py` or `alltests.py`.

**Comment by JukkaL** [comment]
Closing in favor of #1673. Things are already better since pytest can be used for running a subset of tests, though the migration is still incomplete.
